discovery-provider/src/utils/helpers.py

```diff
@@ -73,15 +77,24 @@ def tuple_to_model_dictionary(t, model):
     return dict(zip(keys, t))
 
 
+log_format = {
+    "levelno": "levelno",
+    "level": "levelname",
+    "msg": "message"
+}
+
+formatter = JsonFormatter(
```

Comment:
one more thing that would be super nice would be to pprint this stuff out so stack traces look more readable. right now it looks like
`
{"levelno": 40, "level": "ERROR", "msg": "index_cache.py | Fatal error in main loop\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 160, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py\", line 61, in create_connection\n    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):\n  File \"/usr/local/lib/python3.7/socket.py\", line 748, in getaddrinfo\n    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):\nsocket.gaierror: [Errno -2] Name does not resolve\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 677, in urlopen\n    chunked=chunked,\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 392, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1229, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1275, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1224, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1016, in _send_output\n    self.send(msg)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 956, in send\n    self.connect()\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 187, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 172, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f690a2acf28>: Failed to establish a new connection: [Errno -2] Name does not resolve\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 449, in send\n    timeout=timeout\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 727, in urlopen\n    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py\", line 439, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='audius-identity-service_identity-service_1', port=7000): Max retries exceeded with url: /tracks/trending/day (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f690a2acf28>: Failed to establish a new connection: [Errno -2] Name does not resolve'))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/audius-discovery-provider/src/tasks/generate_trending.py\", line 50, in generate_trending\n    resp = requests.post(identity_trending_endpoint, json=post_body)\n  File \"/usr/local/lib/python3.7/site-packages/requests/api.py\", line 119, in post\n    return request('post', url, data=data, json=json, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/api.py\", line 61, in request\n    return session.request(method=method, url=url, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 530, in request\n    resp = self.send(prep, **send_kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 643, in send\n    r = adapter.send(request, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 516, in send\n    raise ConnectionError(e, request=request)\nrequests.exceptions.ConnectionError: HTTPConnectionPool(host='audius-identity-service_identity-service_1', port=7000): Max retries exceeded with url: /tracks/trending/day (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f690a2acf28>: Failed to establish a new connection: [Errno -2] Name does not resolve'))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/audius-discovery-provider/src/tasks/index_cache.py\", line 50, in update_discovery_cache\n    update_all_trending_cache(self, db, redis)\n  File \"/audius-discovery-provider/src/tasks/index_cache.py\", line 22, in update_all_trending_cache\n    update_trending_cache(self, db, redis, \"day\")\n  File \"/audius-discovery-provider/src/tasks/index_cache.py\", line 12, in update_trending_cache\n    resp = generate_trending(db, time, None, 1000, 0)\n  File \"/audius-discovery-provider/src/tasks/generate_trending.py\", line 55, in generate_trending\n    return api_helpers.error_response(e, 500)\n  File \"/audius-discovery-provider/src/api_helpers.py\", line 31, in error_response\n    return jsonify({'success': False, 'error': error}), error_code\n  File \"/usr/local/lib/python3.7/site-packages/flask/json/__init__.py\", line 309, in jsonify\n    if current_app.config['JSONIFY_PRETTYPRINT_REGULAR'] or current_app.debug:\n  File \"/usr/local/lib/python3.7/site-packages/werkzeug/local.py\", line 347, in __getattr__\n    return getattr(self._get_current_object(), name)\n  File \"/usr/local/lib/python3.7/site-packages/werkzeug/local.py\", line 306, in _get_current_object\n    return self.__local()\n  File \"/usr/local/lib/python3.7/site-packages/flask/globals.py\", line 51, in _find_app\n    raise RuntimeError(_app_ctx_err_msg)\nRuntimeError: Working outside of application context.\n\nThis typically means that you attempted to use functionality that needed\nto interface with the current application object in some way. To solve\nthis, set up an application context with app.app_context().  See the\ndocumentation for more information."}
`

Reply:
loggly actually does a decent job of displaying the stack traces 

<img width="1190" alt="Loggly___Search" src="https://user-images.githubusercontent.com/295938/90198649-1a501200-dd87-11ea-8bcb-1243523064a2.png">

Reply:
Yeah, for sure. So your screenshot if of the error log w/ the JSON Formatter - from the image I manually pushed to staging. This is what is looked like before.
![Screen Shot 2020-08-14 at 8 59 53 AM](https://user-images.githubusercontent.com/7064122/90251996-bb34e080-de0c-11ea-93fb-d6fe6f2f2123.png)